service/matching/taskWriter.go

```diff
@@ -98,12 +104,24 @@ func (w *taskWriter) Start() {
 		return
 	}
 	go func() {
-		if state, err := w.tlMgr.renewLeaseWithRetry(suppressFatalSignal); err == nil {
-			w.taskIDBlock = w.tlMgr.rangeIDToTaskIDBlock(state.rangeID)
-			atomic.StoreInt64(&w.maxReadLevel, w.taskIDBlock.start-1)
-			w.tlMgr.taskAckManager.setAckLevel(state.ackLevel)
-			w.tlMgr.taskReader.Signal()
+		retryForever := backoff.NewExponentialRetryPolicy(1 * time.Second)
+		retryForever.SetMaximumInterval(10 * time.Second)
+		retryForever.SetExpirationInterval(backoff.NoInterval)
+
+		state, err := w.renewLeaseWithRetry(
+			retryForever, common.IsPersistenceTransientError)
+		if err != nil {
+			// because we're retrying transient storage errors forever,
+			// anything else that comes out of here is fatal
+			close(w.shutdownChan)
+			atomic.StoreInt32(&w.status, common.DaemonStatusStopped)
+			w.tlMgr.signalFatalProblem(w.tlMgr.taskQueueID)
+			return
 		}
+		w.taskIDBlock = rangeIDToTaskIDBlock(state.rangeID, w.config.RangeSize)
+		atomic.StoreInt64(&w.maxReadLevel, w.taskIDBlock.start-1)
+		w.tlMgr.taskAckManager.setAckLevel(state.ackLevel)
+		w.tlMgr.taskReader.Signal()
```

Comment:
Can we make this a func? And maybe call it as first line in taskWriterLoop()?

Reply:
Updated as you describe. I wonder if you think it would be better for the loop function to be _just_ the loop and for the init function to be called directly from the goro in Start? That way it's more clear to readers that there's an init phase and then the loop runs. A minor difference tbh.